llvm/lib/Passes/StandardInstrumentations.cpp

```diff
@@ -357,9 +381,11 @@ void ChangeReporter<T>::saveIRBeforePass(Any IR, StringRef PassID,
                                          StringRef PassName) {
   // Is this the initial IR?
   if (InitialIR) {
-    InitialIR = false;
-    if (VerboseMode)
-      handleInitialIR(IR);
+    if (const auto *MF = unwrapIR<MachineFunction>(IR); !MF || !MF->empty()) {
```

Comment:
is the intention here to make the output of instruction selection the "initial IR"?

I feel like being explicit about the fact that instruction selection is actually modifying the MachineFunction is reasonable, so the initial IR being empty isn't too bad. but I'm open to this as well